htsohm/htsohm.py

```diff
@@ -148,55 +148,100 @@ def run_all_simulations(material):
-        material.update_from_dict(results)
-        material.gas_adsorption_bin = calc_bin(
-            abs(material.ga0_absolute_volumetric_loading - material.ga1_absolute_volumetric_loading),
-            *config['gas_adsorption_0']['limits'],
-            config['number_of_convergence_bins']
-        )
+        if config['artificial_data'] == 'off':
+            arguments = [material.run_id, material]
+            if 'helium_void_fraction' in simulations:
+                arguments.append(material.vf_helium_void_fraction)
+            results = simulation.gas_adsorption_0.run(*arguments)
+            material.update_from_dict(results)
+            results = simulation.gas_adsorption_1.run(*arguments)
+            material.update_from_dict(results)
+            material.gas_adsorption_bin = calc_bin(
+                abs(material.ga0_absolute_volumetric_loading - material.ga1_absolute_volumetric_loading),
+                *config['gas_adsorption_0']['limits'],
+                config['number_of_convergence_bins']
+            )
+        else:
+            results = {}
+            results['ga0_absolute_volumetric_loading'] = material.artificial_gas_adsorption()
+            results['ga1_absolute_volumetric_loading'] = material.artificial_gas_adsorption()
+            material.update_from_dict(results)
+            material.gas_adsorption_bin = calc_bin(
+                material.ga0_absolute_volumetric_loading,
+                *config['gas_adsorption_0']['limits'],
+                config['number_of_convergence_bins']
+            )
+
     elif 'gas_adsorption_0' not in simulations:
         material.gas_adsorption_bin = 0
     ############################################################################
     # run surface area simulation
     if 'surface_area' in simulations:
-        results = simulation.surface_area.run(
-                material.run_id, material)
-        material.update_from_dict(results)
-        material.surface_area_bin = calc_bin(
-            material.sa_volumetric_surface_area,
-            *config['surface_area']['limits'],
-            config['number_of_convergence_bins']
-        )
+        if config['artificial_data'] == 'off':
+            results = simulation.surface_area.run(
+                    material.run_id, material)
+            material.update_from_dict(results)
+            material.surface_area_bin = calc_bin(
+                material.sa_volumetric_surface_area,
+                *config['surface_area']['limits'],
+                config['number_of_convergence_bins']
+            )
+        else:
+            results = {}
+            results['sa_volumetric_surface_area'] = material.artificial_surface_area()
+            material.update_from_dict(results)
+            material.surface_area_bin = calc_bin(
+                material.sa_volumetric_surface_area,
+                *config['surface_area']['limits'],
+                config['number_of_convergence_bins']
+            )
+
```

Comment:
I see what you're doing here, and it should work this way.

We ought to be able to replace this whole block of code though with just a config file that lists the tests to run, and then run through them. Let's discuss in person.